### ISSUE TITLE
sqlx's Get doesn't work with single return values

### ISSUE THREAD

**Issue by fiatjaf** [go-cq/cq#27]
I don't know how to phrase this better, so here's an example:

considering

``` go
type Message struct {
    Date      types.NullTime `json:"date"      db:"date"`
    Id        string         `json:"id"        db:"id"`
    From      string         `json:"from"      db:"from"`
    CommentId string         `json:"commentId" db:"commentId"`
}
```

``` go
message := Message{}
err := DB.Get(&message, `
MATCH (m:Message {commentId: {0}})
RETURN m.date AS date,
       m.id AS id,
       m.from as from, m.commentId as commentId`,
        commentId)
```

This works. `message` is populated according to the `Message` definitions.

``` go
message := Message{}
err := DB.Get(&message, `
MATCH (m:Message {commentId: {0}})
RETURN m`,
        commentId)
```

This doesn't work.

---

Now that I wrote, I can imagine why this happens and that it isn't an bug, but perhaps there is a way to make this kind of query easier. Neo4j should have a `RETURN m.*` statement, but it doesn't.

**Comment by fiatjaf** [comment]
I'm confused, the second approach actually works for primary types (`string`, `int`, `bool`), just not for `struct`.

**Comment by fiatjaf** [comment]
Perhaps passing a pointer to a pointer will work? I'll stop commenting.

**Comment by freeeve** [comment]
Yeah, database/sql doesn't understand map types, so there's no way for sqlx to convert from a `map[string]CypherValue` to a struct. I'm thinking for best results we need an underlying API that's not database/sql, which is what I'm planning in v2. Unfortunately I don't think this is fixable in v1.

**Comment by fiatjaf** [comment]
But doesn't analogous cases work with sqlx and pq, for example?

(I'm sorry if this is a stupid question, I'm not complaining, I'm just curious and want to understand more. Also, I'm not sure if I remember seeing this working in analogous cases.)

**Comment by freeeve** [comment]
If you return a table.\* maybe I could see it working.
